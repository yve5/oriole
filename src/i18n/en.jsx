const I18N_DICTIONARY = {
  LANG: 'en',
  LANGUAGE: 'English',
  DATE_FORMAT: 'MM/dd/yyyy',
  LOCALE: 'enUS',
  'About us': 'About us',
  'An error has been detected. Please try again.':
    'An error has been detected. Please try again.',
  Buttons: 'Buttons',
  Changelog: 'Changelog',
  Collapse: 'Collapse',
  Expand: 'Expand',
  Internationalization: 'Internationalization',
  Paging: 'Paging',
  'The request has been sent successfully.':
    'The request has been sent successfully.',
};

export default I18N_DICTIONARY;
