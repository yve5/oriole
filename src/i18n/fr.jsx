const I18N_DICTIONARY = {
  LANG: 'fr',
  LANGUAGE: 'Français',
  DATE_FORMAT: 'dd/MM/yyyy',
  LOCALE: 'fr',
  'About us': 'À propos de nous',
  'An error has been detected. Please try again.':
    'Une erreur a été détectée. Merci de recommencer.',
  Buttons: 'Boutons',
  Changelog: 'Journal des modifications',
  'Changes have been saved.': 'Les changements ont été sauvegardés.',
  'Choose a file': 'Choisir un fichier',
  Collapse: 'Réduire',
  Columns: 'Colonnes',
  'Creation date': 'Date de création',
  'Descending order': 'Ordre décroissant',
  Disabled: 'Désactivé',
  Enabled: 'Activé',
  Expand: 'Développer',
  First: 'Premier',
  'Hello World': 'Bonjour le monde',
  'Hide columns': 'Cacher les colonnes',
  Internationalization: 'Internationalisation',
  Paging: 'Pagination',
  'The request has been sent successfully.':
    'La demande a été envoyée avec succès.',
};

export default I18N_DICTIONARY;
