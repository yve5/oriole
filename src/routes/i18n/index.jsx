import I18nPage from './components/I18n';

export default I18nPage;
