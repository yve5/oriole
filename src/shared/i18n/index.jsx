import LangSelect from './components/LangSelect';
import LangSwitch from './components/LangSwitch';
import I18n from './components/I18n';
import i18nReducer from './reducer';

export { I18n, LangSelect, LangSwitch, i18nReducer };
