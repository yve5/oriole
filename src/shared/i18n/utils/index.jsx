import { checkTermOrder, convertCollectionToArray } from './Term';
import { getInitialState } from './InitialState';
import { checkMissingTerms } from './Missing';
import { formatDate } from './Format';

export {
  formatDate,
  checkTermOrder,
  getInitialState,
  checkMissingTerms,
  convertCollectionToArray,
};
