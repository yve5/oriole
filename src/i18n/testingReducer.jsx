import i18n from '../shared/i18n/reducer';
import fr from './fr';
import en from './en';

export default i18n([fr, en])();
