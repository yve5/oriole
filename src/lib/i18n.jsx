import { I18n, LangSelect, LangSwitch, i18nReducer } from '../shared/i18n';
import {
  trim,
  formatDate,
  checkTermOrder,
  checkMissingTerms,
  convertCollection,
  checkLocalTermOrder,
  convertLocalCollection,
} from '../shared/i18n/utils';

export {
  I18n,
  trim,
  formatDate,
  LangSelect,
  LangSwitch,
  i18nReducer,
  checkTermOrder,
  checkMissingTerms,
  convertCollection,
  checkLocalTermOrder,
  convertLocalCollection,
};
