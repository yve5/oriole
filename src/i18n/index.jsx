import fr from './fr';
import en from './en';
import { tsl } from '../shared/i18n/utils/Translate';

const i18n = {
  tsl: (term, args) => tsl(fr, term, args),
  dictionaries: [fr, en],
  lang: fr.LANG,
};

export default i18n;
