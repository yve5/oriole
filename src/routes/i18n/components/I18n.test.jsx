import React from 'react';
import configureMockStore from 'redux-mock-store';

import { thunk } from 'redux-thunk';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';

import I18nPage from './I18n';
import i18n from '../../../i18n';

const mockStore = configureMockStore([thunk]);

describe('I18n', () => {
  it('should match expected snapshot', async () => {
    const instance = (
      <Provider store={mockStore({ i18n })}>
        <I18nPage />
      </Provider>
    );

    const { asFragment, rerender } = render(instance);

    rerender(instance);

    expect(asFragment()).toMatchSnapshot();
  });
});
