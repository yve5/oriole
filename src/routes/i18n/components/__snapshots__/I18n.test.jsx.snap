// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`I18n > should match expected snapshot 1`] = `
<DocumentFragment>
  <h1>
    Internationalisation
  </h1>
  <select
    class="form-select"
    data-testid="i18n-lang-select"
  >
    <option
      value="fr"
    >
      Français
    </option>
    <option
      value="en"
    >
      English
    </option>
  </select>
  <ul
    class="list-group mt-3"
  >
    <li
      class="list-group-item"
    >
      Pagination
    </li>
    <li
      class="list-group-item"
    >
      Développer
    </li>
    <li
      class="list-group-item"
    >
      Boutons
    </li>
    <li
      class="list-group-item"
    >
      À propos de nous
    </li>
    <li
      class="list-group-item"
    >
      Réduire
    </li>
    <li
      class="list-group-item"
    >
      Journal des modifications
    </li>
    <li
      class="list-group-item"
    >
      Chargement...
    </li>
  </ul>
</DocumentFragment>
`;
